core/store/src/flat/storage.rs

```diff
@@ -122,6 +122,12 @@ impl FlatStorageInner {
             blocks.len(),
             first_height.map(|height| height - flat_head.height),
         );
+        if blocks.len() >= Self::HOPS_LIMIT {
+            let shard_id = self.shard_uid.shard_id();
+            let flat_head_height = flat_head.height;
+            let cached_deltas = self.deltas.len();
+            warn!(target: "chain", shard_id, flat_head_height, cached_deltas, "Flat storage cached deltas exceeded expected limits");
+        }
```

Comment:
No need to check it here, it is covered at least in add_delta -> update_delta_metrics